router/usecase/pools/routable_cw_alloy_transmuter_pool_test.go

```diff
@@ -0,0 +1,253 @@
+package pools_test
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/mocks"
+	"github.com/osmosis-labs/sqs/router/usecase/pools"
+	"github.com/osmosis-labs/sqs/sqsdomain"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+)
+
+func (s *RoutablePoolTestSuite) SetupRoutableAlloyTransmuterPool(tokenInDenom, tokenOutDenom string, balances sdk.Coins) sqsdomain.RoutablePool {
+	cosmwasmPool := s.PrepareCustomTransmuterPool(s.TestAccs[0], []string{tokenInDenom, tokenOutDenom})
+
+	poolType := cosmwasmPool.GetType()
+
+	veryBigNormalizationFactor, _ := osmomath.NewIntFromString("999999999999999999999999999999999999999999")
+
+	mock := &mocks.MockRoutablePool{
+		ChainPoolModel: cosmwasmPool.AsSerializablePool(),
+		CosmWasmPoolModel: sqsdomain.NewCWPoolModel(
+			"crates.io:transmuter", "3.0.0",
+			sqsdomain.CWPoolData{
+				AlloyTransmuter: &sqsdomain.AlloyTransmuterData{
+					AlloyedDenom: ALLUSD,
+					AssetConfigs: []sqsdomain.TransmuterAssetConfig{
+						{Denom: USDC, NormalizationFactor: osmomath.NewInt(100)},
+						{Denom: USDT, NormalizationFactor: osmomath.NewInt(1)},
+						{Denom: "overlypreciseusd", NormalizationFactor: veryBigNormalizationFactor},
+						{Denom: "noprecisionusd", NormalizationFactor: osmomath.ZeroInt()},
```

Comment:
re: the previous comment about constants. I think by having them referenced via shared variables, noting such dependent setup logic would be simpler since we could search variables by references